src/Microsoft.TemplateEngine.TemplateLocalizer.Core/TemplateStringExtractor.cs

```diff
@@ -168,9 +169,9 @@ private void ProcessStringElement(JsonElement element, string elementName, strin
                 return;
             }
 
-            string finalKey = data.KeyPrefix == null ? key : (data.KeyPrefix + "." + key);
+            string finalKey = data.KeyPrefix == null ? key : (data.KeyPrefix + _keySeparator + key);
 
-            if (finalKey.StartsWith(".."))
+            if (finalKey.StartsWith(string.Empty + _keySeparator + _keySeparator))
```

Comment:
nit: Can we find better syntax? :)

Reply:
Updated with a much better alternative. Let me know what you think! I'm also open to suggestions :)